timemachine/md/builders.py

```diff
@@ -34,45 +129,101 @@ def build_protein_system(host_pdbfile: Union[app.PDBFile, str], protein_ff: str,
     modeller = app.Modeller(host_pdb.topology, host_pdb.positions)
     host_coords = strip_units(host_pdb.positions)
 
-    padding = 1.0
-    box_lengths = np.amax(host_coords, axis=0) - np.amin(host_coords, axis=0)
+    water_residues_in_pdb = [residue for residue in host_pdb.topology.residues() if residue.name == WATER_RESIDUE_NAME]
+    num_host_atoms = host_coords.shape[0]
+    if len(water_residues_in_pdb) > 0:
+        host_water_atoms = len(water_residues_in_pdb) * 3
+        # Only consider non-water atoms as the host, does count excipients as the host
+        num_host_atoms = num_host_atoms - host_water_atoms
+        water_indices = np.concatenate([[a.index for a in res.atoms()] for res in water_residues_in_pdb])
+        expected_water_indices = np.arange(host_water_atoms) + num_host_atoms
+        assert np.all(expected_water_indices == water_indices), "Waters in PDB must be at the end of the file"
```

Comment:
nice addition: previously unsafe assumption.

nit: might save a few debugging seconds by printing the mismatch
```suggestion
        np.testing.assert_equal(expected_water_indices, water_indices, err_msg="Waters in PDB must be at the end of the file")
```